main.go

```diff
@@ -15,74 +16,40 @@ import (
 )
 
 var vaultClient *client.Client
+var completerInstance *completer.Completer
 
 var (
 	vshVersion    = ""
 	verbosity     = "INFO"
 	isInteractive = true
 )
 
-func printVersion() {
-	fmt.Println(vshVersion)
+type args struct {
+	CmdString             string `arg:"-c,--cmd" help:"subcommand to run"`
+	DisableAutoCompletion bool   `arg:"--disable-auto-completion" help:"disable auto-completion on paths"`
+	Verbosity             string `arg:"-v,--log-level" help:"DEBUG | INFO | WARN | ERROR - debug option creates vsh_trace.log" default:"INFO" placeholder:"LEVEL"`
 }
 
-func parseInput(line string) (args []string) {
-	quote := '0'
-	escaped := false
-	arg := ""
-
-	for _, c := range line {
```

Comment:
I should've just looked for a library in the first place rather than spending a night trying to implement this. 😎 But I learned I guess.